packages/client/components/GitLabScopingSearchFilterMenu.tsx

```diff
@@ -0,0 +1,160 @@
+                  query {
+                    projects(
+                      ids: null
+                      membership: true
+                      first: 100
+                      sort: "latest_activity_desc"
+                    ) {
+                      edges {
+                        node {
+                          ... on _xGitLabProject {
+                            __typename
+                            id
+                            fullPath
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    `,
+    queryRef,
+    {UNSTABLE_renderPolicy: 'full'}
+  )
+
+  const nullableEdges =
+    query.viewer.teamMember?.integrations.gitlab.api?.query?.projects?.edges ?? []
+  const projects = useMemo(() => getNonNullEdges(nullableEdges).map(({node}) => node), [query])
+  const meeting = query?.viewer?.meeting
+  const meetingId = meeting?.id ?? ''
+  const gitlabSearchQuery = meeting?.gitlabSearchQuery
+  const {selectedProjectsIds} = gitlabSearchQuery!
+  const atmosphere = useAtmosphere()
+
+  const {query: searchQuery, filteredItems: filteredProjects, onQueryChange} = useSearchFilter(
+    projects,
+    getValue
+  )
+  const visibleProjects = filteredProjects.slice(0, MAX_PROJECTS)
+
+  const {portalStatus, isDropdown} = menuProps
+  return (
+    <StyledMenu
+      keepParentFocus
+      ariaLabel='Define the GitLab search query'
+      portalStatus={portalStatus}
+      isDropdown={isDropdown}
+    >
+      <SearchMenuItem
+        placeholder='Search your GitLab projects'
+        onChange={onQueryChange}
+        value={searchQuery}
+      />
+      {visibleProjects.length === 0 && (
+        <EmptyDropdownMenuItemLabel key='no-results'>No projects found!</EmptyDropdownMenuItemLabel>
+      )}
+      {visibleProjects.map((project) => {
+        const {id: projectId, fullPath} = project
+        const isSelected = !!selectedProjectsIds?.includes(projectId)
+
+        const handleClick = () => {
+          commitLocalUpdate(atmosphere, (store) => {
+            const searchQueryId = SearchQueryId.join('gitlab', meetingId)
+            const gitlabSearchQuery = store.get<GitLabSearchQuery>(searchQueryId)!
+            const selectedProjectsIds = gitlabSearchQuery.getValue(
+              'selectedProjectsIds'
+            ) as string[]
+            const newSelectedProjectsIds = isSelected
+              ? selectedProjectsIds.filter((id) => id !== projectId)
+              : [...selectedProjectsIds, projectId]
+            gitlabSearchQuery.setValue(newSelectedProjectsIds, 'selectedProjectsIds')
+          })
+        }
+        return (
+          <MenuItem
+            key={projectId}
+            label={
+              <StyledMenuItemLabel>
+                <StyledCheckBox active={isSelected} />
+                <TypeAheadLabel query={searchQuery} label={fullPath} />
```

Comment:
In GitLab, a project can have subprojects. In the Parabol1 repo, we have several projects that contain several subprojects. So, if we show the `name` of the project in the filter menu, there can be several projects that have the same name. In Parabol1, we have many subprojects called `ppmi`, for example. To differentiate between projects, I think it's best to use `fullPath` instead of `name` in the menu.